src/main/scala/com/fulcrumgenomics/fgsv/util/PutativeBreakpoint.scala

```diff
@@ -5,52 +5,81 @@ package com.fulcrumgenomics.fgsv.util
  * @param rightRefIndex  the reference index (0-based) of the right side of the breakpoint
  * @param rightPos the genomic position (1-based) of the right side of the breakpoint
- * @param sameStrand true if the sides are on the same strand, false otherwise
+ * @param rightPositive true if the sequence on the right hand side of the breakpoint is on the positive strand
  * @param evidence the type of evidence for this breakpoint
  */
-case class PutativeBreakpoint
-(id: Long = 0,
- leftRefIndex: Int,
- leftPos: Int,
- rightRefIndex: Int,
- rightPos: Int,
- sameStrand: Boolean,
- evidence: EvidenceType
-) extends Ordered[PutativeBreakpoint] {
+case class PutativeBreakpoint(id: Long = 0,
+                              leftRefIndex: Int,
+                              leftPos: Int,
+                              leftPositive: Boolean,
+                              rightRefIndex: Int,
+                              rightPos: Int,
+                              rightPositive: Boolean,
+                              evidence: EvidenceType
+                             ) extends Ordered[PutativeBreakpoint] {
   /** Returns true the other breakpoint has the same break ends.  This compares all fields except the evidence type. */
   def sameBreakEnds(that: PutativeBreakpoint): Boolean = {
     this.leftRefIndex == that.leftRefIndex &&
       this.leftPos == that.leftPos &&
+      this.leftPositive == that.leftPositive &&
       this.rightRefIndex == that.rightRefIndex &&
       this.rightPos == that.rightPos &&
-      this.sameStrand == that.sameStrand
+      this.rightPositive == that.rightPositive
   }
 
   /** Defines an ordering over breakpoints, first by genomic coordinates of the left then right end, then evidence type. */
   def compare(that: PutativeBreakpoint): Int = {
     var result: Int = 0
     if (result == 0) result = this.leftRefIndex - that.leftRefIndex
     if (result == 0) result = this.leftPos - that.leftPos
+    if (result == 0) result = this.leftPositive.compare(that.leftPositive)
     if (result == 0) result = this.rightRefIndex - that.rightRefIndex
     if (result == 0) result = this.rightPos - that.rightPos
+    if (result == 0) result = this.rightPositive.compare(that.rightPositive)
     if (result == 0) result = EvidenceType.indexOf(this.evidence) - EvidenceType.indexOf(that.evidence)
     result
   }
+
+  /**
+   * Returns an inverted copy of the breakpoint where the left and right sides of the breakpoint have been
+   * switched.
+   */
+  def inverted: PutativeBreakpoint = copy(
+      leftRefIndex  = rightRefIndex,
+      leftPos       = rightPos,
+      leftPositive  = !rightPositive,
+      rightRefIndex = leftRefIndex,
+      rightPos      = leftPos,
+      rightPositive = !leftPositive
+    )
+
+  /** Returns true if the representation of the breakend is canonical, with the left hand side of the break
+   * earlier on the genome than the right hand side. */
+  def isCanonical: Boolean = (leftRefIndex < rightRefIndex) ||
```

Comment:
I think we should think long and hard about what "canonical" should be, probably review the VCF spec and what callers do.  See the test for the tandem duplication as an example of where we might want to do things differently.